src/main/java/com/parsing/parsers/pdf/parsing/RectangleDetector.java

```diff
@@ -0,0 +1,231 @@
+        return (Math.abs(coordinates[0])) + OFFSET / 2 > distinctPointX && (Math.abs(coordinates[0])) - OFFSET / 2 < distinctPointX;
+    }
+
+    private List<HorizontalLineCoordinate> formHorizontalLinesCoordinates(List<Double> lines, HorizontalLineCoordinate horizontalLineCoordinate) {
+        List<HorizontalLineCoordinate> result = new ArrayList<>();
+        for (Double line : lines) {
+            result.add(new HorizontalLineCoordinate(horizontalLineCoordinate.getLeftPoint(), horizontalLineCoordinate.getRightPoint(), line));
+        }
+        return result;
+    }
+
+    public void saveIMageWithVerticalLines(List<double[]> linesV) {
+        for (double[] point : linesV) {
+            Imgproc.line(verticalLinesMat, new Point(point[0], point[1]), new Point(point[2], point[3]), new Scalar(0, 0, 255), 3, Imgproc.LINE_AA, 0);
+        }
+        Imgcodecs.imwrite("rectVertical.png", verticalLinesMat);
+    }
+
+    public void saveIMageWithVerticalLines2(List<double[]> linesV) {
+        for (double[] point : linesV) {
+            Imgproc.line(verticalLinesMat, new Point(point[0], point[1]), new Point(point[2], point[3]), new Scalar(0, 0, 255), 3, Imgproc.LINE_AA, 0);
+        }
+        Imgcodecs.imwrite("rectVertical2.png", verticalLinesMat);
+    }
+
+    private void saveAllLines(List<VerticalLineCoordinate> verticalLine, List<HorizontalLineCoordinate> horizontalLines) {
+        verticalLine.forEach(
+                x -> Imgproc.line(verticalLinesMat, new Point(x.getXCoordinate(), x.getTopPoint()), new Point(x.getXCoordinate(), x.getBottomPoint()),
+                        new Scalar(0, 0, 255), 3, Imgproc.LINE_AA, 0));
+        horizontalLines.forEach(
+                x -> Imgproc.line(verticalLinesMat, new Point(x.getLeftPoint(), x.getYCoordinate()), new Point(x.getRightPoint(), x.getYCoordinate()),
+                        new Scalar(0, 0, 255), 3, Imgproc.LINE_AA, 0));
+        Imgcodecs.imwrite("All_Lines.png", verticalLinesMat);
+    }
+
+    private List<Double> mergeLines(List<Double> pointsY) {
+        log.debug("Lines has rows at start= " + pointsY.size());
+        List<Double> result = new ArrayList<>(pointsY);
+        for (int i = 0; i < pointsY.size(); i++) {
+            if (i + 1 < pointsY.size()) {
+                double point1 = pointsY.get(i);
+                double point2 = pointsY.get(i + 1);
+                double diff = point2 - point1;
+                if (diff < OFFSET) {
+                    result.remove(point2);
+                }
+            }
+        }
+        log.debug("Lines has rows at finished= " + result.size());
+        return result;
+    }
+
+    private List<Double> sortLinesByY(List<double[]> lines) {
+        List<Double> pointsY = new ArrayList<>();
+        for (double[] line : lines) {
+            pointsY.add(line[1]);
+        }
+        return pointsY.stream().sorted().toList();
+    }
+
+    private double[] sortListArray(List<double[]> lines) {
+        double minY = 50;
```

Comment:
50 is a magic number, create a variable for it please

Reply:
Done.